library/x509_create.c

```diff
@@ -273,10 +273,18 @@ int mbedtls_x509_string_to_names(mbedtls_asn1_named_data **head, const char *nam
         }
 
         if (!in_attr_type && ((*c == ',' && *(c-1) != '\\') || c == end)) {
+            if (*s == '#') {
```

Comment:
Thinking about it, we need to check that there actually is something in the attribute value, otherwise we can pass `CN=,` and cause a 1-byte buffer overread.
```suggestion
            if (s == c) {
                /* Empty AttributeValue */
                mbedtls_free(oid.p);
                return MBEDTLS_ERR_X509_INVALID_NAME;
            } else if (*s == '#') {
```

Reply:
Note: there should be a test for this case